File: drift/src/main/java/drift/com/drift/model/Conversation.java
package drift.com.drift.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import org.jetbrains.annotations.Nullable;

import java.util.Date;

/**
 * Created by eoin on 28/07/2017.
 */

public class Conversation {

    public enum ConversationStatus {OPEN, CLOSED, PENDING}

    @Expose
    @SerializedName("id")
    public Integer id;

    @Expose
    @SerializedName("displayId")
    public Integer displayId;

    @Expose
    @SerializedName("type")
    public String type = "";

    @Expose
    @SerializedName("status")
    public String status;

    @Expose
    @SerializedName("orgId")
    public Integer orgId;

    @Expose
    @SerializedName("inboxId")
    public Integer inboxId;

    @Expose
    @SerializedName("endUserId")
    public Long endUserId;

    @Expose
    @SerializedName("subject")
    public Object subject;

    @Expose
    @SerializedName("preview")
    public String preview;

    @Expose
    @SerializedName("user")
    public User user;

    @Expose
    @SerializedName("updatedAt")
    public Date updatedAt;

    @Nullable
    public ConversationStatus getConversationStatus() {
        try {
            return ConversationStatus.valueOf(status);
        }catch (Throwable t) {
            t.printStackTrace();
            return null;
        }
    }

}

File: drift/src/main/java/drift/com/drift/model/Conversation.kt
package drift.com.drift.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.util.Date

/**
 * Created by eoin on 28/07/2017.
 */

internal class Conversation {

    enum class ConversationStatus {
        OPEN, CLOSED, PENDING;
    }

    @Expose
    @SerializedName("id")
    var id: Long? = null

    @Expose
    @SerializedName("type")
    var type = ""

    @Expose
    @SerializedName("status")
    private var status: String? = null

    @Expose
    @SerializedName("orgId")
    var orgId: Int? = null

    @Expose
    @SerializedName("preview")
    var preview: String? = null

    @Expose
    @SerializedName("updatedAt")
    var updatedAt: Date? = null

    val conversationStatus: ConversationStatus?
        get() {
            return try {
                status?.let { ConversationStatus.valueOf(it) }
            } catch (t: Throwable) {
                null
            }

        }

}
